src/lib/spinel/multi_frame_buffer.hpp

```diff
@@ -371,7 +371,7 @@ template <uint16_t kSize> class MultiFrameBuffer : public FrameWritePointer
 
         aFrame = (aFrame == nullptr) ? mBuffer : aFrame + aLength;
 
-        if (HasSavedFrame() && (aFrame != mWriteFrameStart))
+        if (HasSavedFrame() && (aFrame != mWriteFrameStart) && ((aFrame + kHeaderSize) < GetArrayEnd(mBuffer)))
```

Comment:
Why will there be an incomplete frame while `HasSavedFrame()` is true? Should we avoid saving incomplete frame?

Reply:
The function `GetNextSavedFrame ()`only iterate frames in the multi-frame-buffer. All frames are still be saved in the multi-frame-buffer when calling `GetNextSavedFrame()`, so the function `HasSavedFrame ()` does the right job here.